aggregator/src/aggregator.rs

```diff
@@ -1502,18 +1506,24 @@ impl VdafOps {
                     "Report decryption failed",
                 );
                 upload_decrypt_failure_counter.add(1, &[]);
-                return Ok(());
+                return Err(reject_report(ReportRejectedReason::LeaderDecryptFailure));
             }
         };
 
-        let leader_plaintext_input_share =
-            PlaintextInputShare::get_decoded(&encoded_leader_plaintext_input_share)
-                .map_err(|err| Arc::new(Error::from(err)))?;
+        let decoded_leader_input_share = PlaintextInputShare::get_decoded(
+            &encoded_leader_input_share,
+        )
+        .and_then(|plaintext_input_share| {
+            Ok((
+                plaintext_input_share.extensions().to_vec(),
+                A::InputShare::get_decoded_with_param(
+                    &(&vdaf, Role::Leader.index().unwrap()),
+                    plaintext_input_share.payload(),
+                )?,
+            ))
+        });
 
-        let leader_input_share = match A::InputShare::get_decoded_with_param(
-            &(&vdaf, Role::Leader.index().unwrap()),
-            leader_plaintext_input_share.payload(),
-        ) {
+        let (extensions, leader_input_share) = match decoded_leader_input_share {
```

Comment:
Notice that I've chained these operations together. Previously, for the first decoding phase of turning a decrypted Vec into a `PlaintextInputShare`, we were simply wrapping the error in `aggregator::Error`. 

This seemed incorrect to me. Since it's an error decoding the input share, it is the same class of error as the later `InputShare` decoding operation, where we provide report rejection context and make note of it with an OTEL counter.

I've chained them so that they have the same behavior under error.